queries/period_slippage.sql

```diff
@@ -310,70 +310,139 @@ incoming_and_outgoing_with_buffer_trades as (
     from buffer_trades
 ),
 final_token_balance_sheet as (
-    select solver_address,
-           solver_name,
-           sum(amount) token_imbalance_wei,
-           symbol,
-           token,
-           tx_hash
-    from incoming_and_outgoing_with_buffer_trades
-    group by symbol,
-             token,
-             solver_address,
-             solver_name,
-             tx_hash
+    select
+        solver_address,
+        solver_name,
+        sum(amount) token_imbalance_wei,
+        symbol,
+        token,
+        tx_hash,
+        date_trunc('hour', block_time) as hour
+    from
+        incoming_and_outgoing_with_buffer_trades
+    group by
+        symbol,
+        token,
+        solver_address,
+        solver_name,
+        tx_hash,
+        block_time
+    having
+        sum(amount) != 0
 ),
-
-exact_prices as (
+token_times as (
+    select
+        hour,
+        token
+    from
+        final_token_balance_sheet
+    group by
+        hour,
+        token
+),
+precise_prices as (
     select
-        token as contract_address,
+        contract_address,
         decimals,
-        price
-    from final_token_balance_sheet
-    left join prices.usd
-        on contract_address = token
-        and minute = '{{EndTime}}'
+        date_trunc('hour', minute) as hour,
+        avg(price) as price
+    from
+        prices.usd pusd
+    inner join token_times tt
+        on minute between date(hour) and date(hour) + interval '1 day' 
```

Comment:
we may want to limit this bound to the hour
```suggestion
        on minute between date(hour) and date(hour) + interval '1 hour' 
```

Reply:
I also think 1 hour is sufficient. and then we don't need the last next line

Reply:
on minute between hour and hour + interval '1 hour'

Reply:
We were running into perf issues with this suggestion, will address in a follow up